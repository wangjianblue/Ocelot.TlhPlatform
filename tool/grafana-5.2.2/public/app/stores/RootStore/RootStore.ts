import { types } from 'mobx-state-tree';
import { SearchStore } from './../SearchStore/SearchStore';
import { ServerStatsStore } from './../ServerStatsStore/ServerStatsStore';
import { NavStore } from './../NavStore/NavStore';
import { AlertListStore } from './../AlertListStore/AlertListStore';
import { ViewStore } from './../ViewStore/ViewStore';
import { FolderStore } from './../FolderStore/FolderStore';
import { PermissionsStore } from './../PermissionsStore/PermissionsStore';

export const RootStore = types.model({
  search: types.optional(SearchStore, {
    sections: [],
  }),
  serverStats: types.optional(ServerStatsStore, {
    stats: [],
  }),
  nav: types.optional(NavStore, {}),
  alertList: types.optional(AlertListStore, {
    rules: [],
  }),
  permissions: types.optional(PermissionsStore, {
    fetching: false,
    items: [],
  }),
  view: types.optional(ViewStore, {
    path: '',
    query: {},
    routeParams: {},
  }),
  folder: types.optional(FolderStore, {}),
});

type IRootStoreType = typeof RootStore.Type;
export interface IRootStore extends IRootStoreType {}
