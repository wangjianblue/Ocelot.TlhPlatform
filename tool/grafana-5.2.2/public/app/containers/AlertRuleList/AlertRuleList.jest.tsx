import React from 'react';
import moment from 'moment';
import { AlertRuleList } from './AlertRuleList';
import { RootStore } from 'app/stores/RootStore/RootStore';
import { backendSrv, createNavTree } from 'test/mocks/common';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('AlertRuleList', () => {
  let page, store;

  beforeAll(() => {
    backendSrv.get.mockReturnValue(
      Promise.resolve([
        {
          id: 11,
          dashboardId: 58,
          panelId: 3,
          name: 'Panel Title alert',
          state: 'ok',
          newStateDate: moment()
            .subtract(5, 'minutes')
            .format(),
          evalData: {},
          executionError: '',
          url: 'd/ufkcofof/my-goal',
          canEdit: true,
        },
      ])
    );

    store = RootStore.create(
      {},
      {
        backendSrv: backendSrv,
        navTree: createNavTree('alerting', 'alert-list'),
      }
    );

    page = mount(<AlertRuleList {...store} />);
  });

  it('should call api to get rules', () => {
    expect(backendSrv.get.mock.calls[0][0]).toEqual('/api/alerts');
  });

  it('should render 1 rule', () => {
    page.update();
    let ruleNode = page.find('.alert-rule-item');
    expect(toJson(ruleNode)).toMatchSnapshot();
  });

  it('toggle state should change pause rule if not paused', async () => {
    backendSrv.post.mockReturnValue(
      Promise.resolve({
        state: 'paused',
      })
    );

    page.find('.fa-pause').simulate('click');

    // wait for api call to resolve
    await Promise.resolve();
    page.update();

    expect(store.alertList.rules[0].state).toBe('paused');
    expect(page.find('.fa-play')).toHaveLength(1);
  });
});
