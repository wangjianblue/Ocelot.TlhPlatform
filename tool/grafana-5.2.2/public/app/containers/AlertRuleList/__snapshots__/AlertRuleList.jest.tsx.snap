// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertRuleList should render 1 rule 1`] = `
<li
  className="alert-rule-item"
>
  <span
    className="alert-rule-item__icon alert-state-ok"
  >
    <i
      className="icon-gf icon-gf-online"
    />
  </span>
  <div
    className="alert-rule-item__body"
  >
    <div
      className="alert-rule-item__header"
    >
      <div
        className="alert-rule-item__name"
      >
        <a
          href="d/ufkcofof/my-goal?panelId=3&fullscreen=true&edit=true&tab=alert"
        >
          <Highlighter
            highlightClassName="highlight-search-match"
            searchWords={
              Array [
                "",
              ]
            }
            textToHighlight="Panel Title alert"
          >
            <span>
              <span
                className=""
                key="0"
              >
                Panel Title alert
              </span>
            </span>
          </Highlighter>
        </a>
      </div>
      <div
        className="alert-rule-item__text"
      >
        <span
          className="alert-state-ok"
        >
          <Highlighter
            highlightClassName="highlight-search-match"
            searchWords={
              Array [
                "",
              ]
            }
            textToHighlight="OK"
          >
            <span>
              <span
                className=""
                key="0"
              >
                OK
              </span>
            </span>
          </Highlighter>
        </span>
        <span
          className="alert-rule-item__time"
        >
           for 
          5 minutes
        </span>
      </div>
    </div>
  </div>
  <div
    className="alert-rule-item__actions"
  >
    <button
      className="btn btn-small btn-inverse alert-list__btn width-2"
      onClick={[Function]}
      title="Pausing an alert rule prevents it from executing"
    >
      <i
        className="fa fa-pause"
      />
    </button>
    <a
      className="btn btn-small btn-inverse alert-list__btn width-2"
      href="d/ufkcofof/my-goal?panelId=3&fullscreen=true&edit=true&tab=alert"
      title="Edit alert rule"
    >
      <i
        className="icon-gf icon-gf-settings"
      />
    </a>
  </div>
</li>
`;
